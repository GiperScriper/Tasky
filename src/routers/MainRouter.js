import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import PrivateRoute from './PrivateRoute';
import AsyncComponent from '@/hocs/AsyncComponent';

import Login from '@/components/Login';

const App = AsyncComponent(() => import('@/components/App').then(module => module.default));

const MainRouter = () => (
  <BrowserRouter>
    <Switch>
      <Route path="/login" component={Login} />
      <PrivateRoute path="/" component={App} />
    </Switch>
  </BrowserRouter>
);

export default MainRouter;
