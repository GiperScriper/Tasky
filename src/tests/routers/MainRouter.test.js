import React from 'react';
import { shallow } from 'enzyme';
import MainRouter from '@/routers/MainRouter';

describe('<MainRouter />', () => {
  it('should render MainRouter component correctly', () => {
    const wrapper = shallow(<MainRouter />);
    expect(wrapper).toMatchSnapshot();
  });
});
