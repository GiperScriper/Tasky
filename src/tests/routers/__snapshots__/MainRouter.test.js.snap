// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainRouter /> should render MainRouter component correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      path="/login"
    />
    <PrivateRoute
      component={[Function]}
      path="/"
    />
  </Switch>
</BrowserRouter>
`;
